src/node.c

```diff
@@ -522,8 +522,12 @@ node_get_base_url(nodePtr node)
 {
 	const gchar 	*baseUrl = NULL;
 
-	if (node->subscription)
+	if (node->subscription) {
 		baseUrl = subscription_get_homepage (node->subscription);
+		if (!baseUrl)
+			baseUrl = subscription_get_source (node->subscription);
```

Comment:
Yes, we need the full base URL. And perhaps we shouldn't use the homepage as base URL at all.

In `http://example.com/feed.atom` `example.com` is the authority and `/feed.atom` is the path. 
We need the path part of the base URL to correctly resolve relative links. Relative links can have an absolute path like `/news.html` or a relative path like `news.html`. In the case of a relative path, it must be merged with the base URL path to get the correct link. In your example both resolve to the same thing : `http://example.com/news.html` but if the base URL was `http://example.com/topic/feed.atom` then `/news.html` should resolve to `http://example.com/news.html` and `news.html` should resolve to `http://example.com/topic/news.html`

[Section 5.2 of the RFC3986](https://tools.ietf.org/html/rfc3986#section-5.2) gives the algorithm for resolving relative links. We don't have to implement it ourselves, it is implemented by [xmlBuildURI](http://xmlsoft.org/html/libxml-uri.html#xmlBuildURI). This is why I modified `item_make_link` in item.c to use `common_build_uri` instead of manipulating the string.